.github/workflows/delivery.yml

```
@@ -50,11 +51,35 @@ jobs:
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Build container and push to GitHub Container Registry
+      - name: Build container and export to local Docker
         uses: docker/build-push-action@v5
-        if: ${{ github.event_name == 'release' || github.event_name == 'push' }}
         with:
           context: .
+          load: true
+          tags: local/${{ github.repository }}:scan
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+      - name: Scan Image
+        uses: anchore/scan-action@v3
+        id: scan
+        with:
+          image: local/${{ github.repository }}:scan
+          only-fixed: true
+          fail-build: true
+          severity-cutoff: critical
+          output-format: sarif
+
+      - name: Upload Anchore scan SARIF report
+        uses: github/codeql-action/upload-sarif@v3
+        if: ${{ !cancelled() }}
+        with:
+          sarif_file: ${{ steps.scan.outputs.sarif }}
+
+      - name: Push image to GitHub Container Registry
+        uses: docker/build-push-action@v5
+        if: ${{ github.event_name == 'release' || github.event_name == 'push' }}
```

I would like to suggest that we always push images, except for PR builds, and tag them with a branch reference, much like [in this branch](https://github.com/privacybydesign/irma-saml-bridge/compare/main...chained-session-test).
Would be nice if we can do this for all repo's that build images.

---

Agreed. Something like this I suppose? Will add a note to the original ticket to also apply this to other repos.

```suggestion
        if: ${{ github.event_name != 'pull_request' }}
```

---

@kamphuisem, updated it to also push images with a branch tag